### PR TITLE
storeapi: remove exposure of series

### PR DESCRIPTION
**Pull request by sergiusens**
Series, mostly used with constants.DEFAULT_SERIES has been long deprecated and
has not use.

Remove its broad exposure across the code base. No great changes should be
observed aside from reworking how SnapNotFoundError is implemented by making
use of the new SnapcraftException.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/storeapi/errors.py [hunk 1]
```diff
@@ -111,28 +111,63 @@ def __init__(self, exception):
...
 
-    __FMT_ARCH_CHANNEL = (
-        "Snap {name!r} for {arch!r} cannot be found in the {channel!r} channel."
-    )
-    __FMT_CHANNEL = "Snap {name!r} was not found in the {channel!r} channel."
-    __FMT_SERIES_ARCH = "Snap {name!r} for {arch!r} was not found in {series!r} series."
-    __FMT_SERIES = "Snap {name!r} was not found in {series!r} series."
+        # Defaulting snap_name to "" to support the one case we have that
+        # makes use of snap_id.
+        if snap_name == "" and snap_id is None:
```
Comment (cjp256) [inline snapcraft/storeapi/errors.py]: Just a thought:  you could change the caller w/ snap_id to pass in something like snap_name=f"id {snap_id!r}" and drop the special handling.

File: snapcraft/storeapi/errors.py [hunk 2]
```diff
@@ -111,28 +111,63 @@ def __init__(self, exception):
...
-    def __init__(self, name, channel=None, arch=None, series=None):
-        if channel and arch:
-            self.fmt = self.__FMT_ARCH_CHANNEL
-        elif channel:
-            self.fmt = self.__FMT_CHANNEL
-        elif series and arch:
-            self.fmt = self.__FMT_SERIES_ARCH
-        elif series:
-            self.fmt = self.__FMT_SERIES
+        resolution = f"Ensure you have proper access rights for {snap_identifier!r}."
```
Comment (cjp256) [inline snapcraft/storeapi/errors.py]: Just a though: I'm thinking I tend to use "verify" but here you use "ensure".  We should probably have a template for the errors messages with the common language to improve consistency, whichever way.

Comment (sergiusens) [reply snapcraft/storeapi/errors.py]: This is master (includes docstrings, method names and code comments though):
```
sergiusens@gotham:~/projects/snapcore/snapcraft$ grep -i -R ensure snapcraft  | wc -l
134
sergiusens@gotham:~/projects/snapcore/snapcraft$ grep -i -R verify snapcraft  | wc -l
57
```
I am pretty sure that ensure comes from a design review way back then.

File: snapcraft/storeapi/errors.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft/storeapi/errors.py]: Not sure I am ok with making the internal representation be a reflection of what we want for presentation.

File: tests/unit/store/test_store_client.py [hunk 3]
```diff
@@ -1785,10 +1792,15 @@ def _setup_snap(self):
...
             getattr(self.client, self.attribute),
-            "test-unexistent-snap",
+            "test-nonexistent-snap",
             metadata,
             False,
         )
+
+        self.expectThat(raised._snap_name, Equals("test-nonexistent-snap"))
+        self.expectThat(raised._channel, Is(None))
+        self.expectThat(raised._channel, Is(None))
```
Comment (cjp256) [inline tests/unit/store/test_store_client.py]: Going once, going twice... sold for None! should this be _arch?

File: snapcraft/storeapi/errors.py (hunk 2 above)
Comment (cjp256) [reply snapcraft/storeapi/errors.py]: Sounds good!  I'll try to remember :D

File: snapcraft/storeapi/errors.py (hunk 1 above)
Comment (cjp256) [reply snapcraft/storeapi/errors.py]: Fair enough

**Pull request merged**